csi.proto

```diff
@@ -802,6 +802,18 @@ message ListVolumesRequest {
 message ListVolumesResponse {
   message Entry {
     Volume volume = 1;
+
+    // A list of all `node_id` of nodes that the volume in this entry
+    // is controller published on.
```

Comment:
is a controller-published state more special, in some way, as compared to a node-published state? or node-staged state? reporting additional state here (in the list vols call) is nice, but i'm wondering if we should wrap this in a message type instead so that we have a nicer way to report additional state (if needed) going forward. note the use of wrapper BoolValue types that allow the CO to differentiate between "empty" (SP doesn't know/doesn't report) and `false`.

for example:
```proto
  message PublishContext {
    string node_id = 1; // REQUIRED; reporting *only* a node_id field (and nothing else) doesn't tell the CO anything (and SHOULD NOT happen in practice).
    BoolValue controller_published = 2; // OPTIONAL

    // for future consideration?
    //
    // BoolValue node_staged = 3; // OPTIONAL
    // BoolValue node_published = 4; // OPTIONAL
    // string staging_target_path = 5; // OPTIONAL
    // string target_path = 6; // OPTIONAL
  }

  ...

  // N/A if LIST_VOLUMES_PUBLISH_CTX is not a specified capability.
  repeated PublishContext publish_context = 2;
```

Reply:
I think wrapping in a message type is a great idea in case we want to bundle more information about the Volume in a return in the future. However, I'm not sure that other node related state will be considered in the future in this controller level call. What I would like to propose is actually wrapping the `node_ids` in a `VolumeStatus` message that can contain alternate mutable information about the volume in the future. WDYT?

Reply:
Since `CreateVolume` is a controller operation, it can only know the status of `ControllerPublish` not `NodeStage` or `NodePublish`.

+1 wrapping does seem like a good idea. David's current iteration proposing `VolumeStatus` seems reasonable to me.

@jdef @jieyu @julian-hj @ddebroy thoughts?

Reply:
Yeah, this looks good to me.